Sources/TuistCore/MetadataProviders/XCFrameworkMetadataProvider.swift

```diff
@@ -89,11 +89,25 @@ public final class XCFrameworkMetadataProvider: PrecompiledMetadataProvider, XCF
 
     public func binaryPath(xcframeworkPath: AbsolutePath, libraries: [XCFrameworkInfoPlist.Library]) throws -> AbsolutePath {
         let archs: [BinaryArchitecture] = [.arm64, .x8664]
-        guard let library = libraries.first(where: { !$0.architectures.filter(archs.contains).isEmpty }) else {
+        let binaryName = xcframeworkPath.basenameWithoutExt
+
+        guard let library = libraries.first(where: {
+            let validArchitectures = !$0.architectures.filter(archs.contains).isEmpty
+            guard validArchitectures, let binaryPath = try? path(for: $0, binaryName: binaryName, xcframeworkPath: xcframeworkPath) else { return false }
+            guard localFileSystem.exists(binaryPath) else {
+                let relativePath  = binaryPath.components.suffix(3).joined(separator: "/")
```

Comment:
maybe a comment that describes what the path is expected to be and why it makes sense to reference the last three components?

Reply:
Added a comment for this 👍